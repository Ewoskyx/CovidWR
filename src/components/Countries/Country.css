.country-card {
  position: relative;
  border: none;
}

.country-body {
  background-color: #5788e4;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.country-name {
  background-color: #2d4573;
  box-shadow: rgba(0, 0, 0, 0.35) 0 5px 15px;
  border-radius: 8px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.code-title {
  font-family: 'Cabin', sans-serif;
  font-weight: normal;
}

.card-link {
  color: #dedede;
  font-family: "Lato", sans-serif;
}

.country-population {
  font-family: 'Cabin', sans-serif;
  font-weight: normal;
}
