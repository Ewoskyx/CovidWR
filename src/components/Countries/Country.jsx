import React from 'react';
import PropTypes from 'prop-types';
import './Country.css';
import { NavLink } from 'react-router-dom';
import { BsFillPeopleFill } from 'react-icons/bs';
import { GiWorld } from 'react-icons/gi';

const Country = ({
  name, handleClick, code, population, id,
}) => (

  <div className="col-12 col-md-6 country-card">
    <div className="card-body country-body gap-2">
      <div className="col-12 country-name">
        <h5 className="col-8 code-title p-1">
          <GiWorld />
          {`  ${name}`}
        </h5>
        <h5 className="code-bagde col-4">{code}</h5>
      </div>
      <p className="card-text country-population">
        <BsFillPeopleFill />
        {`  Population: ${population.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`}
      </p>
      <NavLink to="/country" id={id} onClick={handleClick} className="card-link">
        Detailed Report...
      </NavLink>
    </div>
  </div>

);

export default Country;

Country.propTypes = {
  name: PropTypes.string,
  handleClick: PropTypes.func,
  code: PropTypes.string,
  population: PropTypes.string,
  id: PropTypes.string,
};

Country.defaultProps = {
  code: '',
  population: '',
  id: '',
  name: '',
  handleClick: () => {},
};
