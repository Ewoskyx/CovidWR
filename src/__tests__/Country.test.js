import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import Country from '../components/Countries/Country';
import store from '../redux/configStore';

const renderer = require('react-test-renderer');

it('Matches the snapshot of Country component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Country />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Country component', () => {
  render(<Provider store={store}><BrowserRouter><Country /></BrowserRouter></Provider>);
  const element = screen.getByText(/Detailed Report.../i);
  expect(element).toBeInTheDocument();
});
