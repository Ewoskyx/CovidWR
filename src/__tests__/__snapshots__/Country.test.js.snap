// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot of Country component 1`] = `
<div
  className="col-12 col-md-6 country-card"
>
  <div
    className="card-body country-body gap-2"
  >
    <div
      className="col-12 country-name"
    >
      <h5
        className="col-8 code-title p-1"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M253.47 17.406C123.76 17.406 18.437 122.76 18.437 252.47c0 129.707 105.324 235.06 235.03 235.06 129.707 0 235.063-105.353 235.063-235.06 0-129.71-105.355-235.064-235.06-235.064zM367.874 68.75c61.246 38.19 101.97 106.14 101.97 183.72 0 17.143-1.993 33.823-5.75 49.81l-34.25-18.06 22 54.874c-9.454 21.647-22.362 41.432-38 58.687l-43.158-30.936-64.625 47.72-61.656 6.967-13.906-41.78-49.72 26.844-68.093-18.938 9.157 36.594c-28.41-21.793-51.23-50.466-66-83.563L81.25 304.47l32.25 17.124 59.22-9.875 2.843-40.908-37.344-1.718 4.905-17.844 30.78-25.313-25.093-15.625 67.22-38.593-45.345-29.657-66.625 40.187-49.437-15.28c13.812-32.14 35.21-60.22 61.906-82.064l-3.75 44.375 43.376-34.124 72 22.22-22.5-27.407L233 75.562l26.813 28.468 71 9.845-3.5-34.47 41.468 12.657-.905-23.312zm1.156 120.03L278 199.47l28.906 43.218 3.156 64.468L339.25 321l11.438-28.375 62.656 48.656L395.78 294l6.408-48.344-43.75-22.72 10.593-34.155zM221 192.438l-31.594 21.188 36.47 14.78 16.686-14.78L221 192.437zm22.188 144.688l18.687 52.594 19.78-42.564-38.467-10.03z"
          />
        </svg>
          
      </h5>
      <h5
        className="code-bagde col-4"
      >
        
      </h5>
    </div>
    <p
      className="card-text country-population"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1H7zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
        />
        <path
          d="M5.216 14A2.238 2.238 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.325 6.325 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1h4.216z"
          fillRule="evenodd"
        />
        <path
          d="M4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z"
        />
      </svg>
        Population: 
    </p>
    <a
      className="card-link"
      href="/country"
      id=""
      onClick={[Function]}
    >
      Detailed Report...
    </a>
  </div>
</div>
`;
